### ISSUE TITLE
Lets migrate on Symfony directory structure

### ISSUE THREAD

**Issue by evrifaessa** [boxbilling/boxbilling#1262]
### Discussed in https://github.com/boxbilling/boxbilling/discussions/1261

<div type='discussions-op-text'>

<sup>Originally posted by **andpavlenko** April  6, 2022</sup>
What you think about migrate to structure directories same as Symfony Framework?

This could be useful for easy integration of Symfony components in the project.

For good example of project structure see [sylius/sylius-standard](https://github.com/Sylius/Sylius-Standard).</div>

**Comment by andpavlenko** [comment]
The simplest to start could be move:

/src/bb-data/log -> /var/log
/src/bb-data/cache -> /var/cache
/src/bb-data/uploads -> /public/uploads
/src/bb-uploads -> /public/uploads
/src/bb-locale -> /translations
/src/bb-vendor -> /src/vendor
/src/composer.json -> /

Remove all empty index.html. Because this legacy case and don't need with front controller pattern.

This not change anything important in the code base, only paths.

Then register Boxbilling on the packagist.org

**Comment by evrifaessa** [comment]
well, that seems like a nice plan. I can provide you a branch in this repository if you want

**Comment by BenNottelling** [comment]
We would want a good plan for how we want to handle this for an install that's being updated

**Comment by andpavlenko** [comment]
For those who update manually, will need to delete everything (except bb-upload and bb-config.php). Then upload new version.

It's usual way for old sites.

**Comment by evrifaessa** [comment]
I don't think this is a convenient way to handle updates. Our updater system is only capable of downloading and overwriting onto the existing files, we may need to update our updater first.

**Comment by andpavlenko** [comment]
By worrying about backwards compatibility and without global changes you lose the ability to move forward.
This could (or need) be done in branch for major major release 5.x. New features and refactoring need to frozen in stable release branch.

**Comment by andpavlenko** [comment]
> Our updater system is only capable of downloading and overwriting onto the existing files, we may need to update our updater first.

bb-update.php looks like update SQL only and do nothing with files.

**Comment by evrifaessa** [comment]
> bb-update.php looks like update SQL only and do nothing with files.

Talking about this one:
https://github.com/boxbilling/boxbilling/blob/master/src/bb-library/Box/Update.php
